MediaGalleryUi/Model/GetImageDetailsByAssetId.php

```diff
@@ -244,4 +268,28 @@ private function formatImageSize(int $imageSize): string
 
         return sprintf('%sKb', $imageSize / 1000);
     }
+
+    /**
+     * Retrive assets used in the Content
+     *
+     * @param int $assetId
+     * @return array
+     */
+    private function getUsedIn(int $assetId): array
+    {
+        $usedIn = [];
+        try {
+            foreach ($this->assetContentTypes as $assetContentTypeKey => $assetContentType) {
+                $getAssetIds = $this->getAssetsUsedInContentInterface->execute($assetContentTypeKey);
```

Comment:
I think for the performance perspective it would be better to represent a group selection where data will be filtered by the `$assetId` and `contentTypes`: `select main_table.asset_id from main_table.media_content_asset where main_table.asset_id = seetId and main_table.type in (contentTypes)` Also, it looks like a good point to exclude this part from the `GetImageDetailsByAssetId`

Reply:
Yes, You are right. It might create performance issue. So should I add the function in `GetAssetsUsedInContent` class or update the existing one?
Thanks

Reply:
@konarshankar07 I think that for this logic should be implemented in a separate class in MediaGalleryUi module. The `GetAssetUsedIn` sounds good for its name, as for me.

Reply:
Okay got it